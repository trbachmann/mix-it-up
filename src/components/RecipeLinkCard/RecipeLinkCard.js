import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const RecipeLinkCard = (props) => {
  const { recipe } = props;
  let pathToUse;
  (recipe.notes === '') ? (pathToUse = '/desserts/') : (pathToUse = '/my-recipes/')
  return(
    <Link className='RecipeLinkCard--Link' to={pathToUse + recipe.id}>
      <div className='RecipeLinkCard--div'>
        <img className='RecipeLinkCard-img' src={recipe.images[0].hostedLargeUrl} alt={recipe.name}/>
        <h2>{recipe.name}</h2>
        <p>Time: {recipe.totalTime}</p>
        <p>From: {recipe.source.sourceDisplayName}</p>
      </div>
    </Link>
  );
}

RecipeLinkCard.propTypes = {
  recipe: PropTypes.object
}