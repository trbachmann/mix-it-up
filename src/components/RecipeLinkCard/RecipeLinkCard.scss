.RecipeLinkCard--Link {
  text-decoration: none;
  color: black;
}

.RecipeLinkCard--div {
  width: 240px;
  height: 275px;
  border-radius: 8px;
  margin: 0;
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),
  0px 2px 2px 0px rgba(0, 0, 0, 0.14),
  0px 3px 1px -2px rgba(0, 0, 0, 0.12);
  
  &:hover {
    border: 1px solid $darkMaroon;  
  }
  
  p,
  h2 {
    margin-left: 3%;
  }

  h2 {
    font-size: 1.2em;
  }
}

.RecipeLinkCard-img {
  width: 94%;
  border-radius: 8px;
  margin: 3%;
}