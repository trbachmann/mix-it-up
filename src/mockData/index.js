export const mockRecipes = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: ''
  }
];

export const mockRecipe = {
  yield: "serves 16-20",
  totalTime: "30 min",
  images: [
    {
      hostedSmallUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s90",
      hostedMediumUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s180",
      hostedLargeUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s360",
      imageUrlsBySize: {
        90: "https://lh3.googleusercontent.com/TvJfjV21XRXRSrS5qNcbtGRm7VGSTK2GY-bCAQnlodFjP2ZljyTeuUg7dmvR8VnhubINi4wKBQ10wKRGpPRXgDQ=s90-c",
        360: "https://lh3.googleusercontent.com/TvJfjV21XRXRSrS5qNcbtGRm7VGSTK2GY-bCAQnlodFjP2ZljyTeuUg7dmvR8VnhubINi4wKBQ10wKRGpPRXgDQ=s360-c"
      }
    }
  ],
  name: "Frozen Oreo Dessert",
  source: {
    sourceDisplayName: "Creations by Kara",
    sourceSiteUrl: "creationsbykara.com",
    sourceRecipeUrl: "https://www.creationsbykara.com/frozen-oreo-ice-cream-dessert/"
  },
  id: "Frozen-Oreo-Dessert-2621007",
  ingredientLines: [
    "1/4 cup butter, melted",
    "1 package of Oreos, finely crushed (I used my food processor)",
    "1/2 gallon of ice cream (Cookies & Cream & Chocolate Chip Mint are yummy)",
    "Hot fudge sauce"
  ],
  attribution: {
    html: "<a href='http://www.yummly.co/recipe/Frozen-Oreo-Dessert-2621007'>Frozen Oreo Dessert recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
    url: "http://www.yummly.co/recipe/Frozen-Oreo-Dessert-2621007",
    "text": "Frozen Oreo Dessert recipes: information powered by Yummly",
    logo: "https://static.yummly.co/api-logo.png"
  },
  numberOfServings: 18,
  totalTimeInSeconds: 1800,
  attributes: {
    course: [
      "Desserts"
    ]
  },
  flavors: {},
  rating: 4,
  notes: ''
};

export const mockUserRecipeFromStorage = {
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
};

export const mockUserRecipeToChangeNotes = {
  id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
  notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
};

export const mockRecipesWithUserNotes = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
  }
];

export const mockAttribution = { 
  attribution: {
  html: "Recipe search powered by <a href='http://www.yummly.co/recipes'><img alt='Yummly' src='https://static.yummly.co/api-logo.png'/></a>",
  url: "http://www.yummly.co/recipes/",
  text: "Recipe search powered by Yummly",
  logo: "https://static.yummly.co/api-logo.png"
    }
}

export const mockRecipesAfterNoteDelete = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: ''
  }
];

export const mockYummlyResponse = {
  "criteria": {
    "q": "desserts",
    "requirePictures": true,
    "allowedIngredient": null,
    "excludedIngredient": null
  },
  "matches": [
    {
      sourceDisplayName: "Picky Palate",
      ingredientLines: [
        "6 Jumbo marshmallows",
        "6 cups good quality vanilla ice cream",
        "1 cup milk",
        "6 Oreo Cookies"
      ],
      id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      recipeName: "Toasted Marshmallow Cookies and Cream Milkshake",
      attributes: {
        course: [
          "Desserts"
        ]
      }
    }
  ],
  "facetCounts": {},
  "totalMatchCount": 353293,
  "attribution": {
    "html": "Recipe search powered by <a href='http://www.yummly.co/recipes'><img alt='Yummly' src='https://static.yummly.co/api-logo.png'/></a>",
    "url": "http://www.yummly.co/recipes/",
    "text": "Recipe search powered by Yummly",
    "logo": "https://static.yummly.co/api-logo.png"
  }
};

export const mockResultFromCombineMatches = [
  {
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    sourceDisplayName: "Picky Palate",
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ]
  },
  {
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
  }
];