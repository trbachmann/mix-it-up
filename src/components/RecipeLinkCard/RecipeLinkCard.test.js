import React from 'react';
import { RecipeLinkCard } from './RecipeLinkCard';
import { shallow } from 'enzyme';
import * as data from '../../mockData';

describe('RecipeLinkCard', () => {
  it('should match the snapshot when the recipe props has not notes', () => {
    const mockProps = data.mockRecipesWithUserNotes[0];
    const wrapper = shallow(<RecipeLinkCard recipe={mockProps}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the recipe props has notes', () => {
    const mockProps = data.mockRecipesWithUserNotes[1];
    const wrapper = shallow(<RecipeLinkCard recipe={mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});