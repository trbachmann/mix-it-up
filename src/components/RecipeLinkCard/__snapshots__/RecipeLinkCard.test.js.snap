// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeLinkCard should match the snapshot when the recipe props has not notes 1`] = `
<Link
  className="RecipeLinkCard--Link"
  replace={false}
  to="/desserts/Toasted-marshmallow-cookies-and-cream-milkshake-335047"
>
  <div
    className="RecipeLinkCard--div"
  >
    <img
      alt="Toasted Marshmallow Cookies and Cream Milkshake"
      className="RecipeLinkCard-img"
      src="http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360"
    />
    <h2>
      Toasted Marshmallow Cookies and Cream Milkshake
    </h2>
    <p>
      Time: 
      15 min
    </p>
    <p>
      From: 
      Picky Palate
    </p>
  </div>
</Link>
`;

exports[`RecipeLinkCard should match the snapshot when the recipe props has notes 1`] = `
<Link
  className="RecipeLinkCard--Link"
  replace={false}
  to="/my-recipes/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930"
>
  <div
    className="RecipeLinkCard--div"
  >
    <img
      alt="The Best (and Easy) Peanut Butter Cookies"
      className="RecipeLinkCard-img"
      src="https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360"
    />
    <h2>
      The Best (and Easy) Peanut Butter Cookies
    </h2>
    <p>
      Time: 
      40 min
    </p>
    <p>
      From: 
      Julia's Album
    </p>
  </div>
</Link>
`;
